scripts/snapshot-restore-balance/snapshot-balances.js

```diff
@@ -0,0 +1,81 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+const fs = require('fs');
+
+const WS_ENDPOINT = 'wss://testnet-rpc.zkverify.io';
+const DISCARDED_ACCOUNTS = [
+        "5D9txxK9DTvgCznTjJo7q1cxAgmWa83CzHvcz8zhBtLgaLBV",
+        "5ETuZEyLnfVzQCaDM8aQCcsNnz6xjPKvQCtqynCLqwng8QLd",
+        "5D29UEzgStCBTnjKNdkurDNvd3FHePHgTkPEUvjXYvg3brJj",
+        "5DiMVAp8WmFyWAwaTwAr7sU4K3brXcgNCBDbHoBWj3M46PiP",
+        "5EhREncHsntgJaax9YQphk1xN3LxPu2Rzbz4A3g7Ut8cRXWq",
+        "5C84NU2477uHCUF1A8rHb89sP2D2ZsnquPaGa2Htv75FN9gm",
+        "5HdZjrmNAkWQhYQUPNv7YRYnT4vyQswjbNm8eXBvULNQz5wH",
+        "5HjFLKpiCStQgRm6ZM1fT1R5pLKAqQdUG3uh7pvzaQfhdFuB",
+        "5ECktCamcAtBFJirEzvvJmXFxgLMCTAejhqZwLT1Dxn2fwB1",
+        "5EZbvFqx3j7ejqBSPWseif8xL3PwoqMQHdMT8rs9qWoHcdR3"
+    ]
+
+async function takeSnapshot(api, snapshotBalancesFile) {
+    const balances = {};
+    const accounts = await api.query.system.account.keys();
+
+    for (const key of accounts) {
+        const account = key.args[0].toString();
+        if (DISCARDED_ACCOUNTS.includes(account)) continue;
+        const { data: { free: balance } } = await api.query.system.account(account);
+        balances[account] = balance.toString();
+    }
+    fs.writeFileSync(snapshotBalancesFile, JSON.stringify(balances, null, 2));
+
+    console.log('Snapshot saved.');
+}
+
+// Restore balances from the snapshot
+async function restoreBalances(api, keyring, snapshotBalancesFile, custodySeed) {
+    const balances = JSON.parse(fs.readFileSync(snapshotBalancesFile));
+    const custodyAccount = keyring.addFromUri(custodySeed);
+    let nonce = (await api.query.system.account(custodyAccount.address)).nonce.toNumber();
+
+    for (const [account, balance] of Object.entries(balances)) {
+        const transfer = api.tx.balances.transferAllowDeath(account, balance);
```

Comment:
If the existential deposit is increased between the dump and restore, the custody account could be removed before the loop completes. Probably this risk can be negligible if we restore balances in reverse balance order, starting from the one with the least amount. 

Otherwise we might also switch to `transferKeepAlive`, and handle possible errors accordingly.

Reply:
I see what you mean but I guess given the amount of accounts this is very unlikely to happen right ? anyway I will update the takeSnapshot function so that we save in reverse order and that order is respected in the restore one